src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -116,7 +116,12 @@ public class Query : IDisposable
 
             // Process the query into batches 
             BatchParserWrapper parser = new BatchParserWrapper();
-            List<BatchDefinition> parserResult = parser.GetBatches(queryText);
+            ExecutionEngineConditions conditions = null;
+            if (settings.IsSqlCmdMode)
+            {
+                conditions = new ExecutionEngineConditions() { IsSqlCmd = settings.IsSqlCmdMode };
+            }
+            List<BatchDefinition> parserResult = parser.GetBatches(queryText, conditions);
```

Comment:
What happens if settings.IsSqlCmdMode is not true? Are we ok calling parser.GetBatches with parameter conditions as null?

Reply:
The default will continue to be IsSqlCmdMode = false. Yeah, GetBatches handles the conditions being null and follows the simple query creation path.